packages/indicators/src/__tests__/Builder/ArithmeticBuilder/ArithmeticBuilder.test.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { Indicator } from '../../../types';
+import { testBuildAnalytics } from './testBuildAnalytics';
+import { testGetAggregations } from './testGetAggregations';
+import { testGetElementCodes } from './testGetElementCodes';
+import { testConfigValidation } from './testValidateConfig';
+
+jest.mock('../../../Builder/createBuilder', () => ({
+  createBuilder: jest.fn().mockImplementation((indicator: Indicator) => ({
+    getElementCodes: () => indicator.config.elementCodes,
+  })),
+}));
+
+describe('ArithmeticBuilder', () => {
+  describe('validateConfig()', testConfigValidation);
+
+  describe('getAggregations()', testGetAggregations);
+
+  describe('getElementCodes()', testGetElementCodes);
+
+  describe('buildAnalytics()', testBuildAnalytics);
```

Comment:
Apart from `validateConfig`, all other unit tests here explore aspects that are already semi-tested in integration tests. I thought that it's still probably useful (and "better practice"?) to keep the unit tests, since they make it easy to define and test more elaborate scenarios/edge cases.

This decision was made easier by the fact that most of them were already written. I don't know if I would go this way if I had started by writing integration tests in this package. When to use each category seems like is an interesting problem to me, and probably has no specific answers. I'd be happy to hear your thoughts!

Reply:
Also, we don't want to "over-test" a package since tests have maintenance overhead. I tried to be conscious of this but please let me know if you think this is the case here!

For future reference: integration tests probably have minimal maintenance cost and I wish I had them written before starting this refactor! Unit tests required some time to be updated but not too much.

Reply:
I'm keen to keep both, seeing as we have them. In the end, if we find ourselves spending inordinate amounts of time updating unit tests, we can kill them at that point!

For the general case - yeah I agree that there's no definite answer to when one should definitely be used over the other. I favour integration tests for "bang for buck", but unit tests are ideal in situations when there's a specific piece of logic that lives in a single place and is complex/risky/central to the system's successful functioning.